weto/src/main/java/fi/uta/cs/weto/actions/notifications/NotificationActions.java

```diff
@@ -1,110 +1,224 @@
+                        setting.setNotifications(false);
+                    }
+
+                    try {
+                        String emailNotifications = settingsMap.get(setting.getType() + "_emailNotifications");
+                        setting.setEmailNotifications(Boolean.parseBoolean(emailNotifications));
+                    } catch (NullPointerException e) {
+                        setting.setEmailNotifications(false);
+                    }
+
+                    // Uncheck email notifications if notifications are off
+                    if(!setting.isNotifications()) {
+                        setting.setEmailNotifications(false);
+                    }
+
+                    setting.update(courseConnection);
+                }
+            } catch (Exception e) {
+                throw new WetoActionException("Failed to save settings");
+            }
+
+            return SUCCESS;
+        }
+    }
+
+    public static class ViewNotificationCenter extends WetoMasterAction {
+        private ArrayList<Notification> notifications;
+        private HashMap<Integer, String> notificationTypes;
+        private HashMap<Integer, String> courseIdsNames;
+
+        private Integer type;
+        private Integer courseId;
+        private boolean dateDesc;
+        private final String ALLCOURSESOPTION = "All courses";
+        private final String ALLTYPESOPTION = "allTypes";
+
+        // Katso rakentaja vielä.
+        public ViewNotificationCenter() {
+            super();
+            notifications = new ArrayList<>();
+            courseIdsNames = new HashMap<>();
+            notificationTypes = new HashMap<>();
+        }
+
+        public ArrayList<Notification> getNotifications() {
+            return notifications;
+        }
+
+        public void setNotifications(ArrayList<Notification> notifications) {
+            this.notifications = notifications;
+        }
+
+        public HashMap<Integer, String> getNotificationTypes() {
+            return notificationTypes;
+        }
+
+        public void setNotificationTypes(HashMap<Integer, String> notificationTypes) {
+            this.notificationTypes = notificationTypes;
+        }
+
+        public HashMap<Integer, String> getCourseIdsNames() {
+            return courseIdsNames;
+        }
+
+        public void setCourseIdsNames(HashMap<Integer, String> courseIdsNames) {
+            this.courseIdsNames = courseIdsNames;
+        }
+
+        public Integer getType() {
+            return type;
+        }
+
+        public void setType(Integer type) {
+            this.type = type;
+        }
+
+        public Integer getCourseId() {
+            return courseId;
+        }
+
+        public void setCourseId(Integer courseId) {
+            this.courseId = courseId;
+        }
+
+        public boolean getDateDesc() {
+            return dateDesc;
+        }
+
+        public void setDateDesc(boolean dateDesc) {
+            this.dateDesc = dateDesc;
+        }
```

Comment:
Add getter for page title
```suggestion
        }
        
        public String getPageTitle() {
            return pageTitle;
        }
```